apis/v1alpha2/targetallocator_types.go

```diff
@@ -82,6 +90,46 @@ type TargetAllocatorStatus struct {
 	Messages []string `json:"messages,omitempty"`
 }
 
+// ScrapeConfig is a Prometheus scrape config definition.
+type ScrapeConfig map[interface{}]interface{}
```

Comment:
should we use the upstream prometheus type?

Reply:
I thought about it, but that type is actually quite complicated, and we specifically decided not to parse the prometheus receiver config into upstream Prometheus structs in the past. We'd be forced to do that again here.

Reply:
Can it be changed to `map[string]interface{}` which is generally easier to work with

Reply:
I think so. I originally had it as that, but ran into some problems with kubebuilder generation which were ultimately unrelated. Let me try again.